File: springboot/open-api/src/main/java/com/openapi/domain/dto/ws/request/InstructionResultRequest.java
package com.openapi.domain.dto.ws.request;

import com.openapi.domain.constant.ws.Instruction;
import com.openapi.domain.dto.ws.base.CommonResultDto;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 客户端经 {@link com.openapi.domain.constant.ws.WsEvent#INSTRUCTION_RESULT} 上报的单步执行结果。
 */
@EqualsAndHashCode(callSuper = true)
@Data
public class InstructionResultRequest extends CommonResultDto {

    /**
     * 对应 {@link InstructionListResponse} 内 {@code instructions} 中的序号，0-based。
     */
    private Integer index;

    /**
     * 与列表项 {@link com.openapi.domain.dto.ws.response.InstructionTtsItem#getType()} /
     * {@link com.openapi.domain.dto.ws.response.InstructionMcpItem#getType()} 一致。
     */
    private Instruction type;

    /**
     * 业务态，如 success / fail（具体取值由实现约定）。
     */
    private String status;
}

File: app/android/app/src/main/java/com/magicvector/domain/dto/ws/request/InstructionResultRequest.kt
package com.magicvector.domain.dto.ws.request

import com.magicvector.domain.constant.ws.Instruction
import com.magicvector.domain.dto.ws.base.CommonResultDto

/**
 * 客户端经 [com.magicvector.domain.constant.ws.WsEvent.INSTRUCTION_RESULT] 上报的单步执行结果。
 */
data class InstructionResultRequest(
    /** 对应 [com.magicvector.domain.dto.ws.response.InstructionListResponse.instructions] 内元素序号，0-based */
    val index: Int,
    /** 与列表项 [com.magicvector.domain.dto.ws.response.InstructionTtsItem.type] / [InstructionMcpItem.type] 一致 */
    val type: Instruction,
    /** 业务态，如 success / fail */
    val status: String,
    requestId: String = "",
    code: Int = 0,
    message: String = "",
) : CommonResultDto(requestId, code, message)
